### PR TITLE
feat(components/text-editor): add options to 'Create link' (#1808)

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**
[AB#2706830](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2706830)

Contribution from @JosieePerezz

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1844/storybook/)

Component Storybooks:

- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1844/storybooks/text-editor-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1844/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1844/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1844/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1844/playground/)

File: libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts [hunk 1]
```diff
@@ -73,11 +76,13 @@ describe('Text editor', () => {
         CommonModule,
         FormsModule,
         ReactiveFormsModule,
+        SkyTextEditorResourcesModule,
         SkyTextEditorModule,
         RouterTestingModule,
       ],
       declarations: [componentType],
       providers: [
+        { provide: SkyAppResourcesService, useClass: SkyLibResourcesService },
```
Comment (johnhwhite) [inline libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts]: I'm not seeing where this is required. Is it necessary?

Comment (Blackbaud-PaulCrowder) [reply libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts]: Somehow I missed this in my review. I removed it.

File: libs/components/text-editor/src/lib/modules/text-editor/url-modal/text-editor-url-modal.component.ts [hunk 2]
```diff
@@ -72,7 +76,19 @@ export class SkyTextEditorUrlModalComponent {
   readonly #modalContext = inject(SkyUrlModalContext);
   readonly #modalInstance = inject(SkyModalInstance);
 
-  constructor() {
+  constructor(idService: SkyIdService) {
```
Comment (johnhwhite) [inline libs/components/text-editor/src/lib/modules/text-editor/url-modal/text-editor-url-modal.component.ts]: It doesn't look like `idService` is being used.

**Review by johnhwhite — APPROVED** [review]

**Pull request merged**